component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/retriever/MoesifKeyRetriever.java

```diff
@@ -0,0 +1,295 @@
+                } catch (IOException | APICallException e) {
+                    log.error("Retry attempt failed.", e);
+                }
+            }
+            response = null;
+        }
+        return response;
+    }
+
+    /**
+     * Will remove Moesif key corresponding to organization ID.
+     * Existing Moesif SDK client associated with the moesif key will be also removed.
+     *
+     * @param orgID
+     */
+    public void removeMoesifKeyFromMap(String orgID) {
+        String moesifKey = orgIDMoesifKeyMap.remove(orgID);
+        moesifKeyClientMap.remove(moesifKey);
+    }
+
+    /**
+     * Fetches all the available orgID - MoesifKey pairs from the microservice.
+     *
+     * @throws IOException
+     * @throws APICallException
+     */
+    private void callListResource() throws IOException, APICallException {
+        final URL obj;
+        try {
+            obj = new URL(MoesifMicroserviceConstants.LIST_URL);
+        } catch (MalformedURLException ex) {
+            log.error("Failed calling Moesif microservice. Attempted to call url: {}",
+                    MoesifMicroserviceConstants.LIST_URL, ex);
+            return;
+        }
+        String authHeaderValue = getAuthHeader(gaAuthUsername, gaAuthPwd);
+        HttpsURLConnection con = null;
+        try {
+            con = (HttpsURLConnection) obj.openConnection();
+            con.setRequestMethod("GET");
+            con.setRequestProperty("Authorization", authHeaderValue);
+            con.setRequestProperty("Content-Type", MoesifMicroserviceConstants.CONTENT_TYPE);
+            con.setReadTimeout(MoesifMicroserviceConstants.REQUEST_READ_TIMEOUT);
+            int responseCode = con.getResponseCode();
+            if (responseCode == HttpsURLConnection.HTTP_OK) {
+                try (BufferedReader in = new BufferedReader(
+                        new InputStreamReader(con.getInputStream(), StandardCharsets.UTF_8));) {
+                    String inputLine;
+                    StringBuffer response = new StringBuffer();
+
+                    while ((inputLine = in.readLine()) != null) {
+                        response.append(inputLine);
+                    }
+                    updateMap(response.toString());
+                }
+            } else if (responseCode >= 400 && responseCode < 500) {
+                log.error("Getting {} from the microservice.", responseCode);
+            } else {
+                throw new APICallException("Getting " + responseCode + " from the microservice and retrying.");
+            }
+        } finally {
+            if (con != null) {
+                con.disconnect();
+            }
+        }
+    }
+
+    /**
+     * Retrieves moesif key of the given orgID by calling the microservice.
+     *
+     * @param orgID
+     * @return
+     * @throws IOException
+     * @throws APICallException
+     */
+    private String callDetailResource(String orgID) throws IOException, APICallException {
```

Comment:
@RakhithaRR : Shall we add a proper doc comment for this method?

comment : Ack